### DISCUSSION TITLE
Console.log's not working in Production on Vercel

### DISCUSSION THREAD

**Discussion by taylorhlabs** [vercel/next.js#21000] [Help] [answered]
I am trying to debug a backend function in my app that is working fine in my lab but doesn't seem to be working in production. In my Vercel Dashboard, I'm not getting any errors or console.log's in my Functions log. According to [this recent blog post](https://vercel.com/knowledge/where-are-vercel-logs), console.log's should show in the Functions log in the Vercel Dashboard. Am I missing some major import or dependency in production that would cause the function to fail spectacularly? I have Node installed globally on my local machine but I don't need to have Node as a dependency in my project for Vercel, do I? 

My function is below. It fetches an image based on a URL and then tries to upload the image to a Google Cloud Storage bucket. As you can see, I have lots of logging, but none of them show in the Vercel Dashboard.
```
import axios from 'axios';
import { v4 as uuidv4 } from 'uuid';
import { Storage } from '@google-cloud/storage';

export default async function handle(req, res) {
  let { featuredImage } = req.body;

  console.log(featuredImage);

  // generate file name
  const imageName = uuidv4();
  console.log(imageName);

  // download featured image
  let downloaded;
  try {
    downloaded = await axios.get(featuredImage, {
      responseType: 'stream'
    });
  } catch (error) {
    console.log(`Error getting image url: ${error}`);
    // return original featuredImage url if unable to fetch the image directly
    res.json(featuredImage);
    return;
  }

  // check headers for image type
  const fileType = downloaded.headers['content-type'];
  console.log(fileType);
  if (!fileType) {
    console.log('No file type');
    // return empty image since it's likely invalid image format
    res.json('');
    return;
  }
  let imageExt = fileType.split('/')[1];
  console.log(imageExt);
  if (!imageExt) {
    console.log('No file extension');
    // return empty image since it's likely invalid image format
    res.json('');
    return;
  }

  // check if valid image type
  const supportedImageTypes = [ 'apng', 'avif', 'gif', 'jpg', 'jpeg', 'jfif', 'pjpeg', 'pjp', 'png', 'svg', 'webp' ];
  if (!supportedImageTypes.includes(imageExt)) {
    console.log('Not valid file type');
    // return empty image since it's likely invalid image format
    res.json('');
    return;
  }

  // if we made it this far, it's a valid image type
  console.log('Valid file type');

  // get ready to upload stream to GCP Storage
  const storage = new Storage({
    projectId: process.env.PROJECT_ID,
    credentials: {
      client_email: process.env.CLIENT_EMAIL,
      private_key: process.env.PRIVATE_KEY
    }
  });
  const myBucket = storage.bucket(process.env.BUCKET_NAME);
  const file = myBucket.file(`${imageName}.${imageExt}`);

  // upload to GCP Storage
  const uploaded = () => {
    return new Promise((resolve, reject) => {
      downloaded.data.pipe(file.createWriteStream({ gzip: true })).on('finish', resolve).on('error', reject);
    });
  };

  try {
    await uploaded();
  } catch (error) {
    console.log(`Error uploading image to GCP Storage: ${error}`);
    // server error of some kind.
    // if we made it this far, the featuredImage url is probably ok so return that
    res.json(featuredImage);
    return;
  }

  // set new url for image to match new image in GCP Storage
  featuredImage = `https://storage.googleapis.com/my-bucket-365c82f/${imageName}.${imageExt}`;
  res.json(featuredImage);
}
```

**Comment by taylorhlabs** [comment, marked as answer]
Console logging just started working like 15 minutes later. Is there a delay sometimes?

**Comment by roofleDev** [comment]
I've also had issues where the logs don't show up for an uncertain amount of time. I wish there was some insight into this behavior.

**Comment by redochka** [comment]
Same here, not clear what's happening. Sometimes when there is an uncaught error, vercel seems to swallow all logs... few hours later, logs seems to be there again... not reliable!

**Reply by BongBongBang** [reply]
same situation....

**Comment by jgreens** [comment]
@huozhi would it be possible to take a look? Thank you 🙏

**Comment by g-wozniak** [comment]
Same for me, no console.error(), no console.info() or console.log(); winston does not log anything either in Vercel. Sadly, this causes major issues due to inconsistency between local and Vercel environments with Postmark and serverless functions that suddenly stop execution.

**Comment by redochka** [comment]
The bug is here again. Had to deploy and make sure the "use cache" is unchecked and now it's working.

**Comment by joelbqz** [comment]
This bug is still present

**Comment by nvegater** [comment]
please check this.

**Comment by The-Poe** [comment]
Encounter the same issue again

**Comment by ThomasGHenry** [comment]
😭😭😭😭😭😭

**Comment by bask-digital** [comment]
Experiencing this with Nuxt

**Comment by iploooox** [comment]
still here.

**Comment by Yan-Doshchinskiy** [comment]
still here

**Comment by yashb042** [comment]
Still observed

**Reply by wangf1** [reply]
I encounter same issue. Console.log log did not appears in vercel dashboard logs tab.

**Comment by ralph-interworldhighway** [comment]
Still here....

**Comment by yeaung276** [comment]
still here

**Comment by Tegzes** [comment]
still here... extremely annoying

please solve this as soon as possible

**Comment by Alexandregirbal** [comment]
Same here

**Comment by Kettss1** [comment]
same here... this time for a preview

**Comment by sahasrarjn** [comment]
Same here, can't access the realtime logs.

**Comment by hubavka** [comment]
Anyone found a solution to this issue?
Should we use a separate logging plugin?

**Comment by luizlealdev** [comment]
Same to me. Did anyone find a solution?

**Comment by legitapps-25** [comment]
Still here

**Comment by KranzAklilu** [comment]
Damn. Logging is a critical feature on any cloud infrastructure you know :)

**Comment by maribox** [comment]
Found two possible solutions to this:

https://github.com/vercel/next.js/issues/59208#issuecomment-2337302208
https://github.com/vercel/next.js/discussions/68332#discussioncomment-10197608

**Comment by FridayCandour** [comment]
still here in april 18 2025.

**Comment by rovshanr1** [comment]
still here in may 8 2025

**Comment by TradersEdge** [comment]
12.5.2025 still here

**Comment by samconsidine** [comment]
still here

**Comment by cga-ma** [comment]
WTH ? Still here

**Comment by rafadontal** [comment]
still here

**Comment by kwam12** [comment]
still here

**Comment by aryanbaburajan** [comment]
huh. still here?

**Comment by hireshBrem** [comment]
stll here, fr?

**Comment by apostolisCodpal** [comment]
This is quite crippling. I can see logs of api route return statuses, but not `console.log`s that I insert in my code. And this is an api endpoint so I don't see the reason.

**Comment by Will-Copeland** [comment]
Adding my name to the list. Function is failing in production only, fill it with logs, lo and behold nothing to see. 👍 

Whats the point of logging if it doesnt work when something breaks? l o l

**Comment by jkumara** [comment]
November 4th, 2025. Still here.

**Comment by QuizBlogRW** [comment]
Still...

**Comment by noahhai** [comment]
Crazy that this is still a thing.

**Comment by valeriadmitrievna-dev** [comment]
its big 2026 and it's.... still here

**Comment by namishkapoor** [comment]
Current Date 27-03-2026 THE BUG IS STILL THERE!!!!

**Comment by enact-on** [comment]
still
